service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/JobProperties.java

```diff
@@ -272,4 +298,76 @@ JobPropertiesParser toJobPropertiesParser()
 
         return jobPropertiesParser;
     }
+
+    /**
+     * Creates an instance of JobProperties with parameters ready to start an Import job
+     *
+     * @param inputBlobContainerUri URI containing SAS token to a blob container that contains registry data to sync
+     * @param outputBlobContainerUri URI containing SAS token to a blob container.  This is used to output the status of the job and the results.
+     * @return An instance of JobProperties
+     */
+    public static JobProperties createForImportJob(
+            String inputBlobContainerUri,
```

Comment:
why static? wouldn't it vary per instance of the job?

Reply:
This is static as customer does not need to create object of class before accessing this method. I am not sure what you mean as per instance of the job here?

Reply:
Static because one doesn't need an instance to call it, rather it returns an instance. It is a factory method.

Reply:
I changed mine to:

/// <param name="inputBlobContainerUri">URI to a blob container that contains registry data to sync. Including a SAS token is dependent on the <see cref="StorageAuthenticationType" /> parameter.</param>
/// <param name="outputBlobContainerUri">URI to a blob container.  This is used to output the status of the job and the results. Including a SAS token is dependent on the <see cref="StorageAuthenticationType" /> parameter.</param>

Reply:
factory in java...cool. Haven't seen  that in a while

Reply:
won't make you go back and redesign, but in java we usually use builder in these cases. factory is frowned upon.

Reply:
What do people have against factory pattern? Genuinely curious.

Reply:
+1,I also want to know.